packages/edit-post/src/components/header/header-toolbar/index.js

```diff
@@ -96,18 +140,25 @@ function HeaderToolbar() {
 					'Add block',
 					'Generic label for block inserter button'
 				) }
+				showTooltip={ ! showIconLabels }
 			/>
-			{ isLargeViewport && <ToolbarItem as={ ToolSelector } /> }
-			<ToolbarItem as={ EditorHistoryUndo } />
-			<ToolbarItem as={ EditorHistoryRedo } />
-			<ToolbarItem
-				as={ TableOfContents }
-				hasOutlineItemsDisabled={ isTextModeEnabled }
-			/>
-			<ToolbarItem
-				as={ BlockNavigationDropdown }
-				isDisabled={ isTextModeEnabled }
-			/>
+			{ ( isWideViewport || ! showIconLabels ) && overflowItems }
+			{ ! isWideViewport && ! isSmallViewport && showIconLabels && (
+				<Dropdown
```

Comment:
I don't think I can use `DropdownMenu` here because its only permitted children are `MenuItem`, `MenuItemsChoice`, or `MenuGroup`, which don't seem right semantically. Unless I'm missing something?